python/tk_desktop2/action_handler.py

```diff
@@ -437,20 +425,19 @@ def _on_commands_loaded(self, project_id, entity_type, entity_id, link_entity_ty
             # the same configuration. It's silly behavior, but culling the duplicated here
             # is the simplest solution, and works just fine.
             if not self._actions_model.findItems(display_name):
+
+                # Convert the Python Pickle to a JSON string for easier processing from the C++ code
+                pickle_string = command.serialize()
+                pickle_dict = pickle.loads(pickle_string)
+                pickle_dict[self.KEY_PICKLE_STR] = pickle_string
+                json_string = json.dumps(pickle_dict)
+
                 self._actions_model.appendAction(
```

Comment:
Can it be _action_model.appendAction(json_string) ? the display name and to tooltip is in the dict, no ?

Reply:
Yes, actually both of the display name and tooltip are present in the JSON but the Python code relies on the extra param to pass custom display name in '_on_commands_load_failed' but that's easy to change. But now, we're down only to 2 `appendAction` calls.

Note that we'll probably be adding 2 more fields to the JSON: taskId & projectId